youtube_dl/extractor/sdarot.py

```diff
@@ -0,0 +1,308 @@
+        tokens = {}
+        for i in range(episodes_count):
+            episode = i + 1
+            key = getkey(serie, season, episode)
+            token = get_token(self, url, serie, season, episode)
+            tokens[key] = token
+
+        # Free users need to wait 30 seconds before they can get the video link
+        self._sleep(30, video_id, None)
+
+        for i in range(episodes_count):
+            episode = i + 1
+            key = getkey(serie, season, episode)
+            token = tokens[key]
+            info = get_info(self, url, serie, season, episode, token, title)
+            entries.append(info)
+
+        return {
+            '_type': 'playlist',
+            'entries': entries,
+            'id': video_id,
+            'title': title,
+        }
+
+
+class SdarotSerieIE(SdarotSeasonIE):
+    _VALID_URL = r'http:\//(?:www\.)?sdarot\.pm/watch/(?P<serie>\d+)-(?P<title>.*)?/$'
+    _TEST = {
+        'url': 'http://www.sdarot.pm/watch/27-%D7%9E%D7%93-%D7%9E%D7%9F-mad-men/',
+        'info_dict': {
+            'id': '27',
+            'ext': 'mp4',
+            'title': '%D7%9E%D7%93-%D7%9E%D7%9F-mad-men'
+        },
+        'playlist': [],
+        'skip': 'too long for tests'
+    }
+
+    def _real_extract(self, url):
+        m = re.search(self._VALID_URL, url)
+        serie = m.group(1)
+        title = m.group(2)
+
+        video_id = "%s" % serie
+
+        webpage = self._download_webpage(url, video_id)
+        seasons_count = len(re.findall(r'(data-season)', webpage))
+
+        entries = []
+        tokens = {}
+
+        for i in range(seasons_count):
+            season = i + 1
+            season_url = "%s/season/%s" % (url, season)
+            season_webpage = self._download_webpage(season_url, video_id)
+            episodes_count = len(re.findall(r'(data-episode)', season_webpage))
+            for x in range(episodes_count):
+                episode = x + 1
+                token = get_token(self, url, serie, season, episode)
+                key = getkey(serie, season, episode)
+                tokens[key] = token
+
+        # Free users need to wait 30 seconds before they can get the video link
+        self._sleep(30, video_id, None)
+
+        for i in range(seasons_count):
+            season = i + 1
+            season_url = "%s/season/%s" % (url, season)
+            season_webpage = self._download_webpage(season_url, video_id)
+            episodes_count = len(re.findall(r'(data-episode)', season_webpage))
+            for x in range(episodes_count):
+                episode = x + 1
+                key = getkey(serie, season, episode)
+                token = tokens[key]
+                info = get_info(self, url, serie, season, episode, token, title)
+                entries.append(info)
+
+        return {
+            '_type': 'playlist',
+            'entries': entries,
+            'id': video_id,
+            'title': title
+        }
+
+
+def getkey(serie, season, episode):
```

Comment:
OK. How about creating one base class, define the outside functions there, and for playlists use `playlist_result()` so that you reduce boilerplate code